sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -2235,11 +2229,13 @@ public <T> Mono<FeedResponse<T>> readMany(
                                     List<CosmosItemIdentity> list = new ArrayList<>();
                                     list.add(itemIdentity);
                                     partitionRangeItemKeyMap.put(range, list);
+                                    singleItemPartitionRequestMap.put(range, itemIdentity);
                                 } else {
                                     List<CosmosItemIdentity> pairs =
                                         partitionRangeItemKeyMap.get(range);
                                     pairs.add(itemIdentity);
                                     partitionRangeItemKeyMap.put(range, pairs);
+                                    singleItemPartitionRequestMap.remove(range);
```

Comment:
Another alternative is to iterate over the ```partitionRangeItemKeyMap``` and select the single items.  ```getRangeQueryMap``` can be overloaded as one option.

Map put and remove complexes might be lower. The real cost will be around memory. Also abstraction wise it will be a cleaner one IMO.